tests/src/test_parser_yin.c

```diff
@@ -0,0 +1,313 @@
+#include <stdarg.h>
+#include <stddef.h>
+#include <setjmp.h>
+#include <cmocka.h>
+
+#include <stdio.h>
+#include <string.h>
+
+#include "../../src/common.h"
+#include "../../src/tree_schema.h"
+#include "../../src/tree_schema_internal.h"
+#include "../../src/parser_yin.h"
+#include "../../src/xml.h"
+
```

Comment:
Please reuse the logger mechanism from other tests. It hides mostly expected error messages from the output and allows you to check for the specific expected error.